go-controller/pkg/ovn/egressip.go

```diff
@@ -1117,9 +1117,13 @@ func (oc *Controller) generatePodIPCacheForEgressIP(eIPs []interface{}) (map[str
 				continue
 			}
 			for _, pod := range pods {
-				for _, podIP := range pod.Status.PodIPs {
-					ip := net.ParseIP(podIP.IP)
-					egressIPToPodIPCache[egressIP.Name].Insert(ip.String())
+				logicalPort, err := oc.logicalPortCache.get(util.GetLogicalPortName(pod.Namespace, pod.Name))
```

Comment:
using the `logicalPortCache` instead of `pod.Status.PodIPs` makes sense to me, specially since we seem to be doing the same in `addPodEgressIPAssignments` to populate eIPC.podAssignment cache if the pod doesn't exist yet.